toon.py

```diff
@@ -12,31 +12,49 @@
 import homeassistant.helpers.config_validation as cv
 
 # Home Assistant depends on 3rd party packages for API specific code.
-REQUIREMENTS = ['toonlib==0.4.0']
+REQUIREMENTS = ['toonlib==0.5.0']
 
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'toon'
 TOON_HANDLE = 'toon_handle'
+CONF_GAS = 'gas'
+DEFAULT_GAS = True
+CONF_SOLAR = 'solar'
+DEFAULT_SOLAR = False
```

Comment:
But what happens then if you have solar panels but they are not in the initial json? You have no way of knowing whether a person has solar panels, but they just don't show or that they don't show because they are not there? I had issues this morning testing the smoke detectors because they were not in the json. The component set-up where we create the different sensors occurs only once, if the json is incomplete there, we have no way to recover if a more complete json comes along later.